src/analysis/processing/qgsalgorithmurlopener.cpp

```diff
@@ -0,0 +1,87 @@
+/***************************************************************************
+                         qgsalgorithmurlopener.cpp
+                         ---------------------
+    begin                : August 2024
+    copyright            : (C) 2024 by Dave Signer
+    email                : david at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmurlopener.h"
+#include "qgsprocessingparameters.h"
+#include "qgis.h"
+
+#include <QUrl>
+#include <QDesktopServices>
+
+///@cond PRIVATE
+
+QString QgsUrlOpenerAlgorithm::name() const
+{
+  return QStringLiteral( "urlopener" );
```

Comment:
```suggestion
  return QStringLiteral( "openurl" );
```

And rename the class to "QgsOpenUrlAlgorithm". "URL Opener" is a bit too FME-like for me, and doesn't follow the naming convention of our other algorithms. (Eg we use "DXF export", not "DXF exporter", and "Filter by geometry type" not "Geometry type filterer".)